src/http/http_parser.h

```diff
@@ -308,7 +308,7 @@ namespace http
         complete_header();
         auto const& max_headers_count =
           configuration.max_headers_count.value_or(default_max_headers_count);
-        if (headers.size() > max_headers_count)
```

Comment:
This is fixing an off-by-one error in the enforcement of this cap. This is executed each time a new header is identified, and the code above has just popped the previous header if any exists. So if the cap is 256, and `headers.size()` is 256 (ie - we're processing the 257th), we should return an error for _this_ header, not the next one.

Reply:
Is this right? We've just called `complete_header()` that calls `headers.emplace()` (i.e. `headers` size is incremented). So if the cap is 256, and `headers.size()` has just gone up to 256, then we don't want to throw just yet (the [configuration entry](https://microsoft.github.io/CCF/main/operations/configuration.html#http-configuration) is called `max_headers_count`, which implies inclusion of max value). What do you mean by "the code above"?

Reply:
We've just emplaced the _previous_ header (copied out of `partial_parsed_header`). If we're here, and `headers.size()` is 256, then we just emplaced the 256, and we're getting a callback because the parser has found a 257th. The happy path with exactly 256 headers doesn't come here for the 257th, instead it calls `headers_complete()`. So we accept 256, but not 257.

Reply:
Ah! Well spotted and thanks for fixing it.